### PR TITLE
docs: Support semantic preview

### PR DESCRIPTION
**Pull request by zombieJ**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [x] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

ref #46906

### 💡 Background and solution

Kapture 2024-02-04 at 17 10 32

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     -      |
| 🇨🇳 Chinese |    -       |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Review by MadCcc — DISMISSED** [review]

File: .dumi/components/SemanticPreview.tsx [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+
+    return classNames;
+  }, [semantics]);
+
+  const cloneNode = React.cloneElement(children, {
+    classNames: semanticClassNames,
+  });
+
+  // ======================== Hover =========================
+  const MARK_BORDER_SIZE = 2;
```
Comment (li-jia-nan) [inline .dumi/components/SemanticPreview.tsx:35]: 这种常量是不是可以放在组件外面去？

File: .dumi/components/SemanticPreview.tsx [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+                  </Typography.Paragraph>{' '}
+                </Flex>
+              </li>
+            ))}
+          </ul>
+        </Col>
+      </Row>
+
+      <div
+        style={{
```
Comment (li-jia-nan) [inline .dumi/components/SemanticPreview.tsx:147]: inline-style 感觉切换成 classname 比较好呢？

File: .dumi/components/SemanticPreview.tsx (hunk 1 above)
Comment (zombieJ) [reply .dumi/components/SemanticPreview.tsx:35]: 放外面会常驻，放 render 里面就只有在 render 阶段会声明。这个是 Demo 用的，所以特地放里面省点内存。

Comment (li-jia-nan) [reply .dumi/components/SemanticPreview.tsx:35]: 学到了

File: .dumi/components/SemanticPreview.tsx (hunk 2 above)
Comment (zombieJ) [reply .dumi/components/SemanticPreview.tsx:147]: #46768 要换框架的 cssinjs，所以这边全部都先写成 inline 的，免得后期要换掉

File: .dumi/components/SemanticPreview.tsx [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+          position: 'absolute',
+          border: `${MARK_BORDER_SIZE}px solid ${token.colorWarning}`,
+          boxSizing: 'border-box',
+          zIndex: 999999,
+          left: markPos[0] - MARK_BORDER_SIZE,
+          top: markPos[1] - MARK_BORDER_SIZE,
+          width: markPos[2] + MARK_BORDER_SIZE * 2,
+          height: markPos[3] + MARK_BORDER_SIZE * 2,
```
Comment (afc163) [inline .dumi/components/SemanticPreview.tsx:155]: 为啥不用 outline 来画？

Comment (zombieJ) [reply .dumi/components/SemanticPreview.tsx:155]: 可以有个动画效果~

**Review by MadCcc — APPROVED** [review]

**Pull request merged**